dashboard/app/models/user.rb

```diff
@@ -731,7 +731,15 @@ def advertised_scripts
   def in_progress_and_completed_scripts
     backfill_user_scripts if needs_to_backfill_user_scripts?
 
-    user_scripts.compact
+    user_scripts.compact.reject do |user_script|
+      begin
+        user_script.script.nil?
```

Comment:
And then if it's not in the cache, [Script.get_from_cache](https://github.com/code-dot-org/code-dot-org/blob/2bd331c55674584f572158bfad2c48857ecdadf2/dashboard/app/models/script.rb#L283-L287) does a DB lookup.

```ruby
def self.get_from_cache(id)
  return get_without_cache(id) unless should_cache?

  script_cache[id.to_s] || get_without_cache(id)
end
```

so we could directly check `Script.script_cache`, though that might be overkill.